// passkee/core/context/useMiddleware.js
var path = require('path');
//var favicon = require('serve-favicon');
var logger = require('morgan');
var cookieParser = require('cookie-parser');
var bodyParser = require('body-parser');
var express = require('express');
var api=require('../app/api.js');
var cfg= global.config;
//var session= require('./session.js');
var session = require('express-session');
var RedisStore = require('connect-redis')(session);
var useSocket= require('./useSocket.js');

console.log(RedisStore);

module.exports=function(app){
	
	app.use(logger('dev'));
	app.use(bodyParser.json());
	app.use(bodyParser.urlencoded({ extended: false }));
	app.use(cookieParser());
	//app.use((new session(cfg.session)).maker());
	app.use(session({
		store: new RedisStore({
			host: "localhost",
			port: 6379, 
			//db: "passkee"
		}),
		resave:false,
		saveUninitialized:false,
		secret: 'keyboard cat'
	}))
	app.use(useSocket.maker());
	
	app.use(express.static(cfg.static,{
		dotfiles: 'ignore',
		etag: false,
		extensions: ['htm','html','css','png','gif','jpg','js','tpl'],
		index: 'index.html',
		maxAge: '3600000',
		redirect: true,
		setHeaders: function (res, path, stat) {
			res.set('x-timestamp', Date.now())
		}
	}));
	
	/*
	var i=0;
	app.use(function(req, res, next) {
		console.log(i++);
		next();
	});
	*/
	
	/*app.get('/', function(req, res, next) {
		res.sendFile(cfg.static+'/index.html');
	});*/
	
	for(var x in api) if(api[x]) {
		console.log(x);
		app.use(x, api[x]);
	}
	
	
	app.use(function(req, res, next) {
		//if()
		var err = new Error('Not Found');
		err.status = 404;
		next(err);
	});
	
	app.use(function(err, req, res, next) {
		console.log(err);
	});
	
	

	
	
	
}
